Sources/AWSLambdaDeploymentDescriptorGenerator/JSONSChemaReader.swift

```
@@ -44,116 +84,329 @@ enum JSONPrimitiveType: Decodable, Equatable {
+        let minProperties: Int?
+        let maxProperties: Int?
+        
+        
+        // Validate required within string array if present
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            properties = try container.decodeIfPresent([String: JSONUnionType].self, forKey: .properties)
+            minProperties = try container.decodeIfPresent(Int.self, forKey: .minProperties)
+            maxProperties = try container.decodeIfPresent(Int.self, forKey: .maxProperties)
+        }
+    }
+    
     
     // for String
     // https://json-schema.org/understanding-json-schema/reference/string
-    let pattern: String? // or RegEx ?
-    // not supported at the moment
-    // let minLength
-    // let maxLength
-    // let format: String // should be an enum to match specification
+    struct StringSchema: Decodable {
+        let pattern: String?
+        //minLength
+        //maxLength
+        
+        init(pattern: String?){
+            self.pattern = pattern
+        }
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            pattern = try container.decodeIfPresent(String.self, forKey: .pattern)
+        }
+    }
     
     // for Array type
     // https://json-schema.org/understanding-json-schema/reference/array
-    let items: ArrayItem?
-    // not supported at the moment
-    // let prefixItems
-    // let unevaluatedItems
-    // let contains
-    // let minContains
-    // let maxContains
-    // let minItems
-    // let maxItems
-    // let uniqueItems
-
-
-    enum CodingKeys: String, CodingKey {
+    struct ArraySchema: Decodable {
+        let items: JSONType?
+        
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            
+            do {
```

there is no need to use `do` here. Do not handle the error, just use a `try` and let the errors propagate up in the stack